pilot/pkg/model/push_context_test.go

```diff
@@ -110,6 +112,139 @@ func TestMergeUpdateRequest(t *testing.T) {
 	}
 }
 
+func TestAuthNPolicies(t *testing.T) {
+	ps := NewPushContext()
+	env := &Environment{Mesh: &meshconfig.MeshConfig{RootNamespace: "istio-system"}}
+	ps.Env = env
+	authNPolicies := map[string]*authn.Policy{
+		constants.DefaultAuthenticationPolicyName: {},
+
+		"mtls-strict-svc": {
+			Targets: []*authn.TargetSelector{{
+				Name: "mtls-strict-svc",
+			}},
+			Peers: []*authn.PeerAuthenticationMethod{{
+				Params: &authn.PeerAuthenticationMethod_Mtls{},
+			},
+			}},
+
+		"mtls-permissive-svc": {
+			Targets: []*authn.TargetSelector{{
+				Name: "mtls-permissive-svc",
+				Ports: []*authn.PortSelector{
+					{
+						Port: &authn.PortSelector_Number{
+							Number: 80,
```

Comment:
The different of this vs `single-port-mtls` case is that this is using number but the `single-port-mtls-on.yaml` in the integration test is using named port. I assume our current handling is to consult the service definition for the port mapping. @diemtvu do you think this difference can explain the failure?

Can we have a unit test case here as well?

Reply:
Right, it makes sense. We do allow to use port name for matching (https://github.com/istio/istio/blob/master/pilot/pkg/model/config.go#L585)

Reply:
@incfly @diemtvu thanks, I realized the other day due to the failing e2e test from `test.integration.security.kube.presubmit` ReachabilityTest.  Unfortunately it meant refactoring my cache in push_context since it was keyed off of port int.  Just about have those changes ready

Reply:
Great, thanks!

Not sure what's your plan, but maybe we can still key-ed on the int port, and in the cache build-up phase, have additional check the service port name/int mapping.

Reply:
@incfly PTAL a the latest changes. Instead of a `map[hostname][port_number][]*Policy`, I switched to `map[hostname][]Struct{ Port_selector, Policy }`.  Let me know what you think